tests/backup/backup_psa_test.go

```diff
@@ -1,30 +1,596 @@
-// TODO: Remove this file before merging to master
 package tests
 
 import (
+	"fmt"
+	api "github.com/portworx/px-backup-api/pkg/apis/v1"
+	"github.com/portworx/sched-ops/k8s/storage"
+	"github.com/portworx/torpedo/drivers/scheduler/k8s"
+	"golang.org/x/sync/errgroup"
+	v1 "k8s.io/api/core/v1"
+	storageApi "k8s.io/api/storage/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"sync"
+	"time"
+
 	. "github.com/onsi/ginkgo/v2"
+	"github.com/pborman/uuid"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
 	"github.com/portworx/torpedo/pkg/log"
 	. "github.com/portworx/torpedo/tests"
 )
 
-// This is a dummy PSA testcase to validate the PSA related methods for RKE
-var _ = Describe("{DummyPSATestcase}", Label(TestCaseLabelsMap[DummyPSATestcase]...), func() {
+var _ = Describe("{EnableNsAndClusterLevelPSAWithBackupAndRestore}", Label(TestCaseLabelsMap[EnableNsAndClusterLevelPSAWithBackupAndRestore]...), func() {
+	var (
+		backupNames          []string
+		restoreNames         []string
+		appList              = Inst().AppList
+		scheduledAppContexts []*scheduler.Context
+		bkpNamespaces        []string
+		label                map[string]string
+		preRuleNameList      []string
+		postRuleNameList     []string
+		providers            []string
+		sourceScNameList     []string
+		cloudCredName        string
+		cloudCredUID         string
+		backupLocationUID    string
+		backupLocationMap    map[string]string
+		sourceClusterUid     string
+		scName               string
+		params               map[string]string
+		backupNSMap          map[string]string
+		controlChannel       chan string
+		errorGroup           *errgroup.Group
+		backupNamesAllNs     []string
+	)
+	storageClassMapping := make(map[string]string)
+	AppContextsMapping := make(map[string]*scheduler.Context)
+	providers = GetBackupProviders()
+	scheduledAppContexts = make([]*scheduler.Context, 0)
+	bkpNamespaces = make([]string, 0)
+	preRuleNameList = make([]string, 0)
+	postRuleNameList = make([]string, 0)
+	originalList := Inst().AppList
+	label = make(map[string]string)
+	backupLocationMap = make(map[string]string)
+	psaFlag := false
+	backupNSMap = make(map[string]string)
+
 	JustBeforeEach(func() {
-		log.InfoD("No pre-configuration needed")
+		// TODO: Need to update testcase ID
```

Comment:
Added the test case in test rail now
Please update :https://portworx.testrail.net/index.php?/cases/view/299243